src/fswAlgorithms/attGuidance/flybyPoint/flybyPointAlgorithm.h

```diff
@@ -0,0 +1,69 @@
+ ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+ */
+
+#ifndef _FLYBY_POINT_ALGORITHM_H
+#define _FLYBY_POINT_ALGORITHM_H
+
+#include "architecture/msgPayloadDef/AttRefMsgPayload.h"
+#include <Eigen/Dense>
+
+/*! @brief A class to perform flyby pointing */
+class FlybyPointAlgorithm {
+   public:
+    void reset(uint64_t currentSimNanos);
+    AttRefMsgPayload updateState(uint64_t currentSimNanos, Eigen::Vector3d r_BN_N, Eigen::Vector3d v_BN_N);
+    bool checkValidity(uint64_t currentSimNanos, Eigen::Vector3d &r_BN_N, Eigen::Vector3d &v_BN_N) const;
+    void computeFlybyParameters(Eigen::Vector3d &r_BN_N, Eigen::Vector3d &v_BN_N);
+    void computeRN(Eigen::Vector3d &r_BN_N, Eigen::Vector3d &v_BN_N);
+    std::tuple<Eigen::Vector3d, Eigen::Vector3d, Eigen::Vector3d> computeGuidanceSolution() const;
+    double getTimeBetweenFilterData() const;
+    void setTimeBetweenFilterData(double timeBetweenFilterData);
+    double getToleranceForCollinearity() const;
+    void setToleranceForCollinearity(double toleranceForCollinearity);
+    int getSignOfOrbitNormalFrameVector() const;
+    void setSignOfOrbitNormalFrameVector(int signOfOrbitNormalFrameVector);
+    double getMaximumAccelerationThreshold() const;
+    void setMaximumAccelerationThreshold(double maxAccelerationThreshold);
+    double getMaximumRateThreshold() const;
+    void setMaximumRateThreshold(double maxRateThreshold);
+    double getPositionKnowledgeSigma() const;
+    void setPositionKnowledgeSigma(double positionKnowledgeStd);
+
+   private:
+    double dt = 0;                     //!< current time step between last two updates
+    double timeOfFirstRead = 0;        //!< time of first nav solution read
+    double timeBetweenFilterData = 0;  //!< time between two subsequent reads of the filter information
+    double toleranceForCollinearity =
+        0;  //!< tolerance for singular conditions when position and velocity are collinear
+    int signOfOrbitNormalFrameVector = 1;  //!< Sign of orbit normal vector to complete reference frame
+
+    double maxRate = 0;          //!< maximum rate spacecraft can control to, used for validity of solution
+    double maxAcceleration = 0;  //!< maximum acceleration spacecraft can control to, used for validity of solution
+
+    bool firstRead = true;               //!< variable to attest if this is the first read after a Reset
+    double f0 = 0;                       //!< ratio between relative velocity and position norms at time of read [Hz]
+    double gamma0 = 0;                   //!< flight path angle of the spacecraft at time of read [rad]
+    uint64_t lastFilterReadTime = 0;     //!< time of last filter read
+    Eigen::Matrix3d R0N;                 //!< inertial-to-reference DCM at time of read
```

Comment:
:small_red_triangle_down: You can default initialize this matrix as `Eigen::Matrix3d R0N{Eigen::Matrix3d::Identity()};`